app/javascript/src/components/editor/Editor.svelte

```diff
@@ -76,32 +77,37 @@
         if (nullCount) params.args_min_length = v.args.length - nullCount
       }
 
-      if (v.args?.length) {
-        // Add detail arguments in autocomplete results
-        const detail = v.args.map(a => `${ toCapitalize(a.name) }`)
-        const joinedDetail = detail.join(", ")
+      if (!params.args_length) return params
 
-        params.detail_full = joinedDetail
-        params.detail = `(${ joinedDetail.slice(0, 30) }${ joinedDetail.length > 30 ? "..." : "" })`
+      // Add detail arguments in autocomplete results
+      const detail = v.args.map(a => `${ toCapitalize(a.name) }`)
+      const joinedDetail = detail.join(", ")
 
-        // Add apply values when selecting autocomplete, filling in default args
-        const lowercaseDefaults = Object.keys(defaults).map(k => k.toLowerCase())
-        const apply = v.args.map(a => {
-          const string = a.default?.toString().toLowerCase().replaceAll(",", "")
+      params.detail_full = joinedDetail
+      params.detail = `(${ joinedDetail.slice(0, 30) }${ joinedDetail.length > 30 ? "..." : "" })`
 
-          if (lowercaseDefaults.includes(string)) return defaults[toCapitalize(string)]
+      // Add apply values when selecting autocomplete, filling in default args
+      const lowercaseDefaults = Object.keys(defaults).map(k => k.toLowerCase())
+      const useParameterObject = $settings["autocomplete-parameter-objects"] && params.args_length >= $settings["autocomplete-min-parameter-size"]
+      const apply = v.args.map(a => {
+        let string = a.default?.toString().toLowerCase().replaceAll(",", "")
+        if (useParameterObject) string = `\n\t${ a.name }: ${ string }`
```

Comment:
Would it make sense to have a limit on when each parameter is put on a new line and when not?

If I was one of those people that put `autocomplete-min-parameter-size` to 1, maybe I'd like for autocomplete to not expand the parameter object into multiple lines if the parameter amount is <3 or something.

Reply:
Good idea!

![min-newline](https://github.com/Mitcheljager/workshop.codes/assets/12848235/3a567252-8865-4049-9606-adab0364de10)